In [ ]:
# NIM/Nama: 19622217/Lydia Gracia
# Tanggal: 23 November 2022
# Deskripsi: Diberikan data harga rumah yang disewakan di Airbnb di Singapura dalam file P05-03.csv

1. Banyaknya rumah masing-masing neighbourhood.

In [ ]:
import pandas as pd

# KAMUS
# df: dataframe

# pandas membaca file P05-03.csv
df=pd.read_csv("P05-03.csv")

# test read
# pd.read_csv("P05-03.csv")

# mendaftar kemunculan tiap neighbourhood pada data
df["neighbourhood"].value_counts()

Kallang                    1043
Geylang                     994
Novena                      537
Rochor                      536
Outram                      477
Bukit Merah                 470
Downtown Core               428
Bedok                       373
River Valley                362
Queenstown                  266
Tanglin                     210
Singapore River             175
Marine Parade               171
Jurong West                 153
Orchard                     136
Newton                      134
Bukit Timah                 131
Jurong East                 118
Hougang                     109
Clementi                    102
Toa Payoh                   101
Pasir Ris                    71
Serangoon                    69
Sengkang                     67
Woodlands                    67
Bukit Batok                  65
Tampines                     64
Museum                       63
Choa Chu Kang                63
Ang Mo Kio                   58
Bishan                       57
Yishun  

2. Jumlah malam penyewaan minimum terbanyak.

In [ ]:
import pandas as pd

# KAMUS
# df: dataframe

df=pd.read_csv("P05-03.csv")

# mengambil jumlah minimum_nights maksimum
df["minimum_nights"].max()

1000

3. 5 data dengan posisi paling selatan (longitude minimum).

In [ ]:
import pandas as pd

# KAMUS
# df: dataframe

df=pd.read_csv("P05-03.csv")

# mengurutkan data berdasarkan longitude menaik
# kemudian mengambil 5 data pertama
df.sort_values(["longitude"], ascending=[True])[:5]

id                                             name    host_id  \
2978  20791161  YOUR entire PRIVATE LUXURY PENTHOUSE condo unit  122991242   
1655  13756029                 Master room 151 Ang Mo Kio Ave 5   76690273   
2767  19758047                                      Ryan's Home  128313510   
1430  12261620      Fresh & modern 2 bedroom suite opposite NTU    4673970   
3335  21953185             Nearest to NTU Singapore(南洋理工大学附近房间)  147505789   

       host_name neighbourhood_group            neighbourhood  latitude  \
2978          Jj         West Region                     Tuas   1.31909   
1655  光雄 Terence         West Region  Western Water Catchment   1.40620   
2767         Fan         West Region  Western Water Catchment   1.35424   
1430      Pamela         West Region              Jurong West   1.34110   
3335      Albert         West Region              Jurong West   1.34127   

      longitude        room_type  price  minimum_nights  number_of_reviews  \
2978  103.64656  Entire home/apt  10000               2                  5   
1655  103.66547  Entire home/apt     60             180                  2   
2767  103.68162     Private room     31               1                  1   
1430  103.68520  Entire home/apt     81             180                  0   
3335  103.68536     Private room     50             180                  0   

     last_review  reviews_per_month  calculated_host_listings_count  \
2978  2017-10-31               0.21                               1   
1655  2016-07-07               0.05                               1   
2767  2017-07-12               0.04                               1   
1430         NaN                NaN                               1   
3335         NaN                NaN                               1   

      availability_365  
2978                89  
1655               364  
2767                 0  
1430                 0  
3335               365

4. Rumah yang ada di neighbourhood group West Region.

In [ ]:
import pandas as pd

# KAMUS
# df: dataframe

df=pd.read_csv("P05-03.csv")

# mengambil data rumah yang berada di neighbourhood group West Region
df.loc[df["neighbourhood_group"]=="West Region"]["name"]


24                  Comfortable Condo Room - Boon Lay MRT
58      S$950/mth spacious room for short/long term lease
89                         Cozy room, convenient location
105                       Student Guardianship & Homestay
122                   ★2+1condo next to MRT, mall & more★
                              ...                        
7752                        near Clementi MRT female only
7782               西部文理有保安的公寓楼下就是地铁站+巴士总站+商场｡南洋理工大学NUS附近｡
7856               Spacious 3 bedroom at Perfect location
7867                           Nearby NUS big common room
7900                                              环境优雅的公寓
Name: name, Length: 540, dtype: object

5. Koefisien korelasi antara banyak review di aplikasi dengan banyaknya hari ketersediaan rumah dalam
satu tahun. Apakah berkorelasi lurus, terbalik, atau tidak berkolerasi? Jelaskan.

In [ ]:
import pandas as pd

# KAMUS
# df: dataframe

df=pd.read_csv("P05-03.csv")

# mengambil koefisien korelasi antara number_of_reviews dengan availability_365
df["number_of_reviews"].corr(df["availability_365"])

-0.06665806434499989

Dibrerikan korelasi -0.06665806434499989 mendekati 0, artinya banyak review di aplikasi dan banyaknya hari ketersediaan rumah dalam satu tahun tidak berkorelasi